src/Octoshift/GithubClient.cs

```diff
@@ -111,27 +114,27 @@ public virtual async IAsyncEnumerable<JToken> GetAllAsync(string url)
         public virtual async Task<string> DeleteAsync(string url) => (await SendAsync(HttpMethod.Delete, url)).Content;
 
         private async Task<(string Content, KeyValuePair<string, IEnumerable<string>>[] ResponseHeaders)> SendAsync(
-            HttpMethod httpMethod, string url, object body = null, HttpStatusCode status = HttpStatusCode.OK)
+            HttpMethod httpMethod, string url, object body = null, HttpStatusCode status = HttpStatusCode.OK, string acceptHeader = null)
         {
             url = url?.Replace(" ", "%20");
-
             _log.LogVerbose($"HTTP {httpMethod}: {url}");
 
-            if (body != null)
+            using var request = new HttpRequestMessage(httpMethod, new Uri(url));
+            
+            if (acceptHeader != null)
             {
-                _log.LogVerbose($"HTTP BODY: {body.ToJson()}");
+                request.Headers.Accept.Add(MediaTypeWithQualityHeaderValue.Parse(acceptHeader));    
             }
 
-            using var payload = body?.ToJson().ToStringContent();
-            using var response = httpMethod.ToString() switch
+            if (body != null)
             {
-                "GET" => await _httpClient.GetAsync(url),
-                "DELETE" => await _httpClient.DeleteAsync(url),
-                "POST" => await _httpClient.PostAsync(url, payload),
-                "PUT" => await _httpClient.PutAsync(url, payload),
-                "PATCH" => await _httpClient.PatchAsync(url, payload),
-                _ => throw new ArgumentOutOfRangeException($"{httpMethod} is not supported.")
-            };
+                _log.LogVerbose($"HTTP BODY: {body.ToJson()}");
+                
+                var payload = body.ToJson();
+                request.Content = new StringContent(payload, Encoding.UTF8, "application/json");
```

Comment:
We already have an extension for this. Also since we are instantiating the `HttpContent` we need to dispose it:
```c#
using var payload = body?.ToJson().ToStringContent();
request.Content = payload;
```

Reply:
Don't we lose the encoding and the media type of the content when doing this?

Reply:
I can see that the headers in the request automatically set to JSON after making the change.